### ISSUE TITLE
App needs to run with Python 3

### ISSUE THREAD

**Issue by ligiabernardet** [ufs-community/ufs-mrweather-app#177]
One of the upgrades for this release is the CCPP capability to use Python 3 (implemented by @climbfuji , which was requested by users. The standalone UFS Weather Model has been successfully tested with Python 3.  However, @pjpegion has reported that the he was not able to run the App with Python 3. 
```
File "/work/noaa/gsienkf/ppegion/ufs-mrweather-app/src/model/FV3/cime/cime_config/buildnml", line 190, in prep_input
    for tile in xrange(1,7):
NameError: name 'xrange' is not defined
```
It appears xrange is Python 2.x; may need to switch to range. 
Attn: @uturuncoglu

**Comment by uturuncoglu** [comment]
@ligiabernardet i fix the issue and replace all xrange statements with range. It is in ufs_update_11 branch of UFSATM interface now. Once the module files are updated in the model side, I'll update the app and let you know to test.

**Comment by ligiabernardet** [comment]
@uturuncoglu Now that the machine xml is updated, are there any updates needed on the App? Or can this issue be closed?

**Comment by climbfuji** [comment]
The instructions we got for testing are using a branch `ufs-release-v1.1` for the app. The naming convention that we agreed upon however says that we should be working on branch `release/public-v1`. The documentation updates all went into branch `release/public-v1`. I assume that ufs-release-v1.1 will be merged into `release/public-v1` and the release tag `ufs-v1.1.0` will be created from that branch. We should do the final testing from that branch, too.

**Comment by ligiabernardet** [comment]
@climbfuji Are you saying you'd like a PR to release/public-v1 before we start testing? Or test and, after successful, do the PR?

**Comment by climbfuji** [comment]
We can do testing now, but then we need to redo it after updating `release/public-v1`. Either way is fine. By the way, according to https://ufs-mrweather-app.readthedocs.io/en/latest/quickstart.html#setup-the-environment, the only two platforms that have central=recommended `UFS_INPUT` directories are cheyenne and hera.

**Comment by climbfuji** [comment]
@uturuncoglu I am probably doing something stupid, but I am getting the following error on my Mac:
```
dom.heinzeller@heinzeller-lt:~/scratch/ufs-mrweather-app/ufs-mrweather-app-release-public-v1/llvm [llvmmpionlydev]> ./cime/scripts/create_newcase --case c96_gfsv15p2_llvm --compset GFSv15p2 --res C96 --workflow ufs-mrweather --machine macos
Compset longname is FCST_ufsatm%v15p2_SLND_SICE_SOCN_SROF_SGLC_SWAV
Compset specification file is /Users/dom.heinzeller/scratch/ufs-mrweather-app/ufs-mrweather-app-release-public-v1/llvm/src/model/FV3/cime/cime_config/config_compsets.xml
ERROR: Makes no sense to have empty read-only file: /Users/dom.heinzeller/scratch/ufs-mrweather-app/ufs-mrweather-app-release-public-v1/llvm/cime/src/drivers/nuopc/cime_config/config_component.xml
```
Do you see what I am doing wrong? Thanks ...

**Comment by uturuncoglu** [comment]
@climbfuji could you try with setting following?

export UFS_DRIVER=nems

**Comment by uturuncoglu** [comment]
@ligiabernardet it seems that this issue is resolved. @pjpegion Could you test it again.

**Comment by climbfuji** [comment]
> @climbfuji could you try with setting following?
> 
> export UFS_DRIVER=nems

Yes, that works. Note that this is not in the ufs mrw readthedocs (at least not for version 1.0.0). Thanks!

**Comment by climbfuji** [comment]
@ligiabernardet do we have documentation on how to download the data? I can't run `./case.build` on my mac, because the automatic downloads are no longer supported:
```
Building case in directory /Users/dom.heinzeller/scratch/ufs-mrweather-app/ufs-mrweather-app-release-public-v1/llvm/c96_gfsv15p2_llvm
sharedlib_only is False
model_only is False
Generating component namelists as part of build
Creating component namelists
  2020-08-31 16:03:01 atm
   Calling /Users/dom.heinzeller/scratch/ufs-mrweather-app/ufs-mrweather-app-release-public-v1/llvm/src/model/FV3/cime/cime_config/buildnml
Checking /Users/dom.heinzeller/scratch/ufs-mrweather-app/UFS_INPUT/ufs_inputdata/icfiles/201908/20190829 directory to find raw input files ...
No input file/s found locally!
Please download raw initial condition and rename the file/s using following convention:
	 NEMSIO: atm.input.ic.nemsio and sfc.input.ic.nemsio
	 GRIB2 : atm.input.ic.grb2
	 NetCDF: atm.input.ic.nc and sfc.input.ic.nc
ERROR: Exiting ...
```

**Comment by uturuncoglu** [comment]
@climbfuji @ligiabernardet yes, that environment variable is new and requires to support multiple UFS app using single interface and CIME version. I could come up with better solution in the future but it is requires for now and the documentation needs to reflect this change and also others such as input data naming convention, disabling automatic download for initial conditions etc.

**Comment by pjpegion** [comment]
@uturuncoglu CIME and the model now work with python3,. I am now testing with python2 to make sure that still works.

**Comment by ligiabernardet** [comment]
This work is complete and ready for further testing.

**Comment by ligiabernardet** [comment]
Closing
